sklearn/ensemble/_forest.py

```diff
@@ -181,9 +181,23 @@ def _parallel_build_trees(
         elif class_weight == "balanced_subsample":
             curr_sample_weight *= compute_sample_weight("balanced", y, indices=indices)
 
-        tree.fit(X, y, sample_weight=curr_sample_weight, check_input=False)
+        # since the deprecation warning has been showed in the ensemble class,
+        # we ignore the warnings raise within individual trees
+        with catch_warnings():
+            filterwarnings(
```

Comment:
Instead of catching the warning here, could we update `_make_estimator` to correctly set `max_features` to avoid the warning? We do something similar there for `criterion`:

https://github.com/scikit-learn/scikit-learn/blob/abbee570f31a91243c22b1892e42056bb915c056/sklearn/ensemble/_base.py#L164-L168

Reply:
Updated the code as suggested :)